Source/WebCore/svg/SVGAnimationElement.cpp

```diff
@@ -430,8 +430,9 @@ float SVGAnimationElement::calculatePercentFromKeyPoints(float percent) const
 
     ASSERT(!m_keyPoints.isEmpty());
     ASSERT(calcMode() != CalcMode::Paced || animationMode() == AnimationMode::Path);
-    ASSERT(keyTimes.size() > 1);
     ASSERT(m_keyPoints.size() == keyTimes.size());
+    if (keyTimes.size() <= 1)
+        return percent;
```

Comment:
I do not think this is the right approach to fix this bug. According to the [specs](https://svgwg.org/specs/animations/#AnimateMotionElement) and in the keyPoints section:

> If a list of ‘[keyPoints](https://svgwg.org/specs/animations/#KeyPointsAttribute)’ is specified, there must be exactly as many values in the ‘[keyPoints](https://svgwg.org/specs/animations/#KeyPointsAttribute)’ list as in the ‘[keyTimes](https://svgwg.org/specs/animations/#KeyTimesAttribute)’ list.

> If there are any errors in the ‘[keyPoints](https://svgwg.org/specs/animations/#KeyPointsAttribute)’ specification (bad values, too many or too few values), then the document is in error (see [Error processing](https://svgwg.org/svg2-draft/implnote.html#ErrorProcessing)).

The good news is `SVGAnimationElement::startedActiveInterval()` validates the values of the animation attributes and sets the `m_animationValid` accordingly. So all we need is to make sure `m_animationValid` to set to false if `keyPoints` and `keyTimes` don't match. If we do this then there is no need to change this function at all.

To do this I think you will need to change this line in `SVGAnimationElement::startedActiveInterval()`

```
m_animationValid = calcMode == CalcMode::Paced || !hasAttributeWithoutSynchronization(SVGNames::keyPointsAttr) || (keyTimes.size() > 1 && keyTimes.size() == m_keyPoints.size());
```

to be 

```
m_animationValid = !hasAttributeWithoutSynchronization(SVGNames::keyPointsAttr) || (keyTimes.size() > 1 && keyTimes.size() == m_keyPoints.size());
```

The reason for removing this clause is having `animationMode == AnimationMode::Path` makes `calcMode == CalcMode::Paced` to be ignored and we end up using the `keyPoints`. So we have to ignore `calcMode == CalcMode::Paced` when we check the validaty the attributes of `animationMode == AnimationMode::Path`.

Reply:
That does seem to work. PR updated.